scripts-dev/check-newsfragment

```diff
@@ -42,8 +42,8 @@ echo "--------------------------"
 echo
 
 matched=0
-for f in $(git diff --name-only FETCH_HEAD... -- changelog.d); do
-    # check that any modified newsfiles on this branch end with a full stop.
+for f in $(git diff --diff-filter=d --name-only FETCH_HEAD... -- changelog.d); do
```

Comment:
`--diff-filter=d` tells git to ignore deleted files here. This should probably be a separate PR but at this stage I've got about 15 of the things open :-S